Assets/Scripts/Enemy/Enemy.cs

```diff
@@ -0,0 +1,184 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Enemy : MonoBehaviour
+{
+    [SerializeField] private int maxHealth = 20;
+    [SerializeField] private Rigidbody2D _rigidBody;
+    [SerializeField] private Behavior enemyBehavior;
+    [SerializeField] private Transform agreCircle;
+    [SerializeField] private EnemyMovement movementController;
+    [SerializeField] private float agreRange = 4.3f;
+    [SerializeField] private float activeAgreRange = 5f;
+    [SerializeField] private GameObject dropSoul;
+    [SerializeField] private int damage = 20;
+    [SerializeField] private float verticalKnockback;
+    [SerializeField] private float horizontalKnockback;
+    [SerializeField] private float cancelMovementTime;
+    private GameObject[] player;
+    private Behavior startEnemyBehavior;
+    private int currentHealth;
+    private bool isTakingDamage;
+    private bool isDead;
+    public GameObject GetPlayer
+    {
+        get
+        {
+            return player[0];
+        }
+    }
```

Comment:
Follow conventions. Add lines between methods 
Refactor ur code(just naming and smth like adding or deleting empty lines)